db/background_mgr.go

```diff
@@ -52,53 +85,181 @@ type BackgroundManagerStatus struct {
 // BackgroundManagerProcessI is an interface satisfied by any of the background processes
 // Examples of this: ReSync, Compaction
 type BackgroundManagerProcessI interface {
-	Run(options map[string]interface{}, terminator chan struct{}) error
+	Init(options map[string]interface{}, clusterStatus []byte) error
+	Run(options map[string]interface{}, persistClusterStatusCallback updateStatusClusterAwareCallback, terminator chan struct{}) error
 	GetProcessStatus(status BackgroundManagerStatus) ([]byte, error)
 	ResetStatus()
 }
 
+type updateStatusClusterAwareCallback func() error
```

Comment:
What does 'ClusterAware' signify in this naming?  Any reason it can't just be updateStatusCallbackFunc?

(elsewhere I think we follow the naming pattern of the Func suffix for function signature alias types)

Reply:
Naming was just because it doesn't actually do anything in the non-cluster aware 'mode'. But I can drop that then I guess theoretically it could have other stuff added to it in the future if required.